apps/src/templates/teacherHomepage/CollapsibleSection.jsx

```diff
@@ -0,0 +1,114 @@
+import React from 'react';
+import FontAwesome from '../FontAwesome';
+import color from "../../util/color";
+
+const styles = {
+  section: {
+    width: 1600,
+  },
+  heading: {
+    paddingLeft: 20,
+    paddingRight: 10,
+    paddingTop: 10,
+    paddingBottom: 20,
+    marginLeft: 15,
+    fontSize: 24,
+    fontFamily: '"Gotham 3r", sans-serif',
+    zIndex: 2,
+    color: color.charcoal,
+    width: 1075
+  },
+  arrowIcon: {
+    paddingRight: 8
+  },
+  linkToViewAll: {
+    color: color.teal,
+    fontSize: 12,
+    fontFamily: '"Gotham", sans-serif',
+    fontWeight: 'bold',
+    marginTop: -2,
+    display: 'inline'
+  },
+  chevron: {
+    display: 'inline',
+    color: color.teal,
+    fontSize: 10,
+    fontWeight: 'bold',
+    marginLeft: 15,
+  },
+  linkBox: {
+    display: 'inline',
+    float: 'right',
+    textDecoration: 'none'
+  },
+  content: {
+    marginBottom: 50,
+    marginLeft: 10
+  }
+};
+
+const CollapsibleSection = React.createClass({
+  propTypes: {
+    children: React.PropTypes.oneOfType([
+      React.PropTypes.node,
+      React.PropTypes.arrayOf(React.PropTypes.node)
+    ]),
+    header: React.PropTypes.string.isRequired,
+    linkText: React.PropTypes.string.isRequired,
+    link: React.PropTypes.string.isRequired,
+  },
+
+  getInitialState() {
+    return {open: true};
+  },
+
+  toggleContent() {
+    this.setState({open: !this.state.open});
+  },
+
+  renderContent() {
+    const content = this.props.children;
+
+    if (this.state.open) {
+      if (content.length > 1) {
+        return (
+          <div style={styles.content}>
+             {content.slice(0,2).map((course, index) =>
+               <div style={styles.card} key={index}>
+                 {course}
+               </div>
+             )}
+          </div>
+        );
+      }
+      return (
+        <div style={styles.content}>
+          {content}
+        </div>
+      );
+    }
```

Comment:
Suggestion:
You first figure out the set of items you want to display, and make it an array.
```jsx
const childItems = _.isArray(this.props.children) ? this.props.children.slice(0, 2) : [this.props.children];
```
Now I think you can have a much simpler return
```jsx
return (
  <div style={styles.content}>
    {childElements.map((course, index) => 
      <div style={styles.card} key={index}>
        {course}
      </div>
    }
  </div>
);